### PR TITLE
[WIP] Upgrading Django Template to 2.0+

### LINKED ISSUE #3418: Django 2.0 template issues
WIP list of the things that don't work when using Django 2.0 rc1

Starter sample:

error when logging in: 'WSGIRequest' object has no attribute 'session'

Polls sample:

`on_delete` parameter is now required for ForeignKey (param was present but optional since 1.9)
ModuleNotFound for: `from django.core.urlresolvers import reverse`

### LINKED ISSUE #4807: Django analysis support for 'path'
New in Django 2
`path()` is used by the templates instead of `url()` so we'll need to analyze those

https://github.com/Microsoft/PTVS/blob/f7dfa91bf0d37d065d3e45f0f4f86bf8fe22c150/Python/Product/Django.Analysis/DjangoAnalyzer.cs#L230

### PR DESCRIPTION
**Pull request by RaymonGulati1**
Fix #3418 
Fix #4807 

Updating the templates to work with Django 2.0+

### CODE REVIEW

**Comment by RaymonGulati1** [comment]
Since this is a work in progress, these are only initial changes. They have yet to go through extensive testing and verification. 
The changes for the "Polls Django Web Project" will be submitted shortly.

File: Python/Templates/Django/ProjectTemplates/Python/Web/WebRoleDjango/urls.py [hunk 1]
```diff
@@ -1,21 +1,20 @@
...
+    Function views
+        1. Add an import:  from my_app import views
+        2. Add a URL to urlpatterns:  path('', views.home, name='home')
+    Class-based views
+        1. Add an import:  from other_app.views import Home
+        2. Add a URL to urlpatterns:  path('', Home.as_view(), name='home')
+    Including another URLconf
+        1. Import the include() function: from django.urls import include, path
+        2. Add a URL to urlpatterns:  path('blog/', include('blog.urls'))
+"""
```
Comment (huguesv) [inline Python/Templates/Django/ProjectTemplates/Python/Web/WebRoleDjango/urls.py]: Add the admin:
```
from django.contrib import admin
from django.urls import path

urlpatterns = [
    path('admin/', admin.site.urls),
]
```

File: Python/Templates/Django/ProjectTemplates/Python/Web/WebRoleDjango/manage.py [hunk 2]
```diff
@@ -1,17 +1,24 @@
...
+if __name__ == '__main__':
     os.environ.setdefault(
-        "DJANGO_SETTINGS_MODULE",
-        "$safeprojectname$.settings"
-    )
-
-    from django.core.management import execute_from_command_line
-
+        'DJANGO_SETTINGS_MODULE', 
+        '$safeprojectname$.settings.settings')
```
Comment (huguesv) [inline Python/Templates/Django/ProjectTemplates/Python/Web/WebRoleDjango/manage.py]: same comment as above

File: Python/Templates/Django/ProjectTemplates/Python/Web/StarterDjangoProject/manage.py [hunk 3]
```diff
@@ -1,17 +1,24 @@
...
-if __name__ == "__main__":
+if __name__ == '__main__':
     os.environ.setdefault(
-        "DJANGO_SETTINGS_MODULE",
-        "$safeprojectname$.settings"
-    )
-
-    from django.core.management import execute_from_command_line
-
+        'DJANGO_SETTINGS_MODULE',
```
Comment (huguesv) [inline Python/Templates/Django/ProjectTemplates/Python/Web/StarterDjangoProject/manage.py]: same comment as above

File: Python/Templates/Django/ProjectTemplates/Python/Web/DjangoProject/manage.py [hunk 4]
```diff
@@ -1,17 +1,24 @@
...
+if __name__ == '__main__':
     os.environ.setdefault(
-        "DJANGO_SETTINGS_MODULE",
-        "$safeprojectname$.settings"
-    )
-
-    from django.core.management import execute_from_command_line
-
+        'DJANGO_SETTINGS_MODULE', 
+        '$safeprojectname$.settings.settings')
```
[the remark this answers is not in the document]
Comment (RaymonGulati1) [reply Python/Templates/Django/ProjectTemplates/Python/Web/DjangoProject/manage.py]: Modified--see latest changes

File: Python/Templates/Django/ProjectTemplates/Python/Web/DjangoProject/urls.py [hunk 5]
```diff
@@ -1,21 +1,22 @@
...
-    # Examples:
-    # url(r'^$', $safeprojectname$.views.home, name='home'),
-    # url(r'^$safeprojectname$/', include('$safeprojectname$.$safeprojectname$.urls')),
-
-    # Uncomment the admin/doc line below to enable admin documentation:
-    # url(r'^admin/doc/', include('django.contrib.admindocs.urls')),
-
-    # Uncomment the next line to enable the admin:
-    # url(r'^admin/', include(admin.site.urls)),
+
```
[the remark this answers is not in the document]
Comment (RaymonGulati1) [reply Python/Templates/Django/ProjectTemplates/Python/Web/DjangoProject/urls.py]: Modified--see latest changes

File: Python/Templates/Django/ProjectTemplates/Python/Web/WebRoleDjango/urls.py (hunk 1 above)
Comment (RaymonGulati1) [reply Python/Templates/Django/ProjectTemplates/Python/Web/WebRoleDjango/urls.py]: Modified--see latest changes

File: Python/Templates/Django/ProjectTemplates/Python/Web/WebRoleDjango/manage.py (hunk 2 above)
Comment (RaymonGulati1) [reply Python/Templates/Django/ProjectTemplates/Python/Web/WebRoleDjango/manage.py]: Modified--see latest changes

File: Python/Templates/Django/ProjectTemplates/Python/Web/StarterDjangoProject/manage.py (hunk 3 above)
Comment (RaymonGulati1) [reply Python/Templates/Django/ProjectTemplates/Python/Web/StarterDjangoProject/manage.py]: Modified--see latest changes

**Pull request merged**